gui-backend/src/main/java/com/rln/gui/backend/implementation/GuiBackendApiImplementation.java

```diff
@@ -158,21 +177,24 @@ public void postWallets(@Valid @NotNull WalletDTO walletDTO) {
     // Get all the balances for all addresses in a wallet
     @Override
     public List<Balance> getBalances(Long walletId) {
-        if (!walletId.equals(ONLY_SUPPORTED_WALLET.getId()))
+        if (ONLY_SUPPORTED_WALLET_ID != walletId) {
```

Comment:
This will break if `walletId` is `null`:
```
jshell> 1l == null
|  Error:
|  bad operand types for binary operator '=='
|    first type:  long
|    second type: <nulltype>
|  1l == null
|  ^--------^
```
Best would be to make `walletId` primitive.

Reply:
how bad...
unfortunately I don't think we can make it primitive (the API has a Long)
what about
1. making ONLY_SUPPORTED_WALLET_ID a Long and using its equals
2. or using Object.equals?

Reply:
I see. I guess your suggestion 1 would be most appropriate then.